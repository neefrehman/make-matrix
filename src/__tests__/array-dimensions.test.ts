import makeMatrix from "../index";

// Test arrays of specific dimensions, without any initialised values
test("length: 1 test - No initialValues", () => {
    expect(makeMatrix([1])).toMatchSnapshot();
});

test("length: 2 test - No initialValues", () => {
    expect(makeMatrix([2])).toMatchSnapshot();
});

test("length: 7 test - No initialValues", () => {
    expect(makeMatrix([7])).toMatchSnapshot();
});

test("1x1 test - No initialValues", () => {
    expect(makeMatrix([1, 1])).toMatchSnapshot();
});

test("1x4 test - No initialValues", () => {
    expect(makeMatrix([2, 4])).toMatchSnapshot();
});

test("4x11 test - No initialValues", () => {
    expect(makeMatrix([4, 11])).toMatchSnapshot();
});

test("1x1x1 test - No initialValues", () => {
    expect(makeMatrix([1, 1, 1])).toMatchSnapshot();
});

test("3x3x3 test - No initialValues", () => {
    expect(makeMatrix([3, 2, 3])).toMatchSnapshot();
});

test("1x1x1x1 test - No initialValues", () => {
    expect(makeMatrix([1, 1, 1, 1])).toMatchSnapshot();
});

test("1x4x5x2 test - No initialValues", () => {
    expect(makeMatrix([1, 4, 5, 2])).toMatchSnapshot();
});

test("1x1x1x1x1 test - No initialValues", () => {
    expect(makeMatrix([1, 1, 1, 1, 1])).toMatchSnapshot();
});

test("2x6x2x5x2 test - No initialValues", () => {
    expect(makeMatrix([2, 6, 2, 5, 2])).toMatchSnapshot();
});

test("1x1x1x1x1x1 test - No initialValues", () => {
    expect(makeMatrix([1, 1, 1, 1, 1, 1])).toMatchSnapshot();
});

test("2x4x1x5x3x1 test - No initialValues", () => {
    expect(makeMatrix([2, 4, 1, 5, 5, 1])).toMatchSnapshot();
});
