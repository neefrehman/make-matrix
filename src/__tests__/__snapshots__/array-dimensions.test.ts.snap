// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1x1 test - No initialValues 1`] = `
Array [
  Array [
    null,
  ],
]
`;

exports[`1x1x1 test - No initialValues 1`] = `
Array [
  Array [
    Array [
      null,
    ],
  ],
]
`;

exports[`1x1x1x1 test - No initialValues 1`] = `
Array [
  Array [
    Array [
      Array [
        null,
      ],
    ],
  ],
]
`;

exports[`1x1x1x1x1 test - No initialValues 1`] = `
Array [
  Array [
    Array [
      Array [
        Array [
          null,
        ],
      ],
    ],
  ],
]
`;

exports[`1x1x1x1x1x1 test - No initialValues 1`] = `
Array [
  Array [
    Array [
      Array [
        Array [
          Array [
            null,
          ],
        ],
      ],
    ],
  ],
]
`;

exports[`1x4 test - No initialValues 1`] = `
Array [
  Array [
    null,
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    null,
  ],
]
`;

exports[`1x4x5x2 test - No initialValues 1`] = `
Array [
  Array [
    Array [
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
    ],
    Array [
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
    ],
    Array [
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
    ],
    Array [
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
      Array [
        null,
        null,
      ],
    ],
  ],
]
`;

exports[`2x4x1x5x3x1 test - No initialValues 1`] = `
Array [
  Array [
    Array [
      Array [
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
      ],
    ],
    Array [
      Array [
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
      ],
    ],
    Array [
      Array [
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
      ],
    ],
    Array [
      Array [
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
      ],
    ],
    Array [
      Array [
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
      ],
    ],
    Array [
      Array [
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
      ],
    ],
    Array [
      Array [
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
      ],
    ],
  ],
]
`;

exports[`2x6x2x5x2 test - No initialValues 1`] = `
Array [
  Array [
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
      Array [
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
        Array [
          null,
          null,
        ],
      ],
    ],
  ],
]
`;

exports[`3x3x3 test - No initialValues 1`] = `
Array [
  Array [
    Array [
      null,
      null,
      null,
    ],
    Array [
      null,
      null,
      null,
    ],
  ],
  Array [
    Array [
      null,
      null,
      null,
    ],
    Array [
      null,
      null,
      null,
    ],
  ],
  Array [
    Array [
      null,
      null,
      null,
    ],
    Array [
      null,
      null,
      null,
    ],
  ],
]
`;

exports[`4x11 test - No initialValues 1`] = `
Array [
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
]
`;

exports[`length: 1 test - No initialValues 1`] = `
Array [
  null,
]
`;

exports[`length: 2 test - No initialValues 1`] = `
Array [
  null,
  null,
]
`;

exports[`length: 7 test - No initialValues 1`] = `
Array [
  null,
  null,
  null,
  null,
  null,
  null,
  null,
]
`;
